pkg/planner/core/operator/logicalop/logical_datasource.go

```diff
@@ -167,6 +171,18 @@ func (ds *DataSource) PredicatePushDown(predicates []expression.Expression, opt
 	}
 	ds.PushedDownConds, predicates = expression.PushDownExprs(util.GetPushDownCtx(ds.SCtx()), predicates, kv.UnSpecified)
 	appendDataSourcePredicatePushDownTraceStep(ds, opt)
+	if ds.SCtx().HasFTSFunc() {
```

Comment:
I think this flag should not be at the session level.
If this SQL involves more than 1 table and only one of them involves FTS function, then `DataSource` of all tables will go to this branch.